src/main/java/gtPlusPlus/xmod/gregtech/api/metatileentity/implementations/base/GregtechMeta_MultiBlockBase.java

```diff
@@ -566,7 +568,7 @@ public int canBufferOutputs(ItemStack[] aItemOutputs, FluidStack[] aFluidOutputs
                                     FlexiblePair<ItemStack, Integer> t =
                                             new FlexiblePair<ItemStack, Integer>(u.getKey(), aRemainder);
                                     // Remove the 'input' stack from the recipe outputs, so we don't try count it again.
-                                    aInputMap.remove(u);
+                                    iterator.remove();
                                     // Add the remainder stack.
                                     aInputMap.add(t);
```

Comment:
I'm pretty sure that addition will throw CME.
Maybe that continue should continue bus iteration